### ISSUE TITLE
Memory Leak?

### ISSUE THREAD

**Issue by travisghansen** [feedbin/refresher#6]
I run a small personal instance of feedbin and refresher seems to have no end to the amount of memory it consumes.  If I simply kill it and restart all the memory is released with 0 perceived changes in behavior (aside from less memory consumption).

Any thoughts or tips?

**Comment by benubois** [comment]
Hi @travisghansen,

That's certainly possible. However, I'm most likely not going to look into it anytime soon. So I have two thoughts.

1. It would be amazing if you wanted to look into this. There's two articles I like that guide you through looking at what ruby is doing with memory. [Debugging memory leaks in Ruby](https://samsaffron.com/archive/2015/03/31/debugging-memory-leaks-in-ruby) and [Ruby 2.1: objspace.so](http://tmm1.net/ruby21-objspace/). You can access the refresher environment by running `bundle exec rake console` in the project dir. If you go this route I'm happy to help out.
2. Restart the process periodically with cron. Considerably less work and less elegant, but will solve the problem.

Thanks!

**Comment by travisghansen** [comment]
@benubois for option #2 please forgive my ignorance but what's the proper options I can add to ```bundle exec foreman start``` to get it to write a pid file?

**Comment by benubois** [comment]
Hi @travisghansen,

foreman is a helper for managing multiple processes, but it is not necessary to use refresher. The [Procfile](https://github.com/feedbin/refresher/blob/master/Procfile) defines what processes it should manage. In the case of refresher it is starting Sidekiq.

Sidekiq has options you can pass to it including a `--pidfile` option. So to control where the pidfile goes you could run something like the following in the project directory:

```bash
bundle exec sidekiq -c 1 -q feed_refresher_fetcher_critical,2 -q feed_refresher_fetcher -r ./app/boot.rb --pidfile /tmp/sidekiq.pid
```

Please let me know if that makes sense.

Thanks!

**Comment by travisghansen** [comment]
Yes, seems to work.  However I've updated both the app and refresher to latest master and now refresher doesn't seem to be doing anything.  Is there a missing migration perhaps (I did run db:migrate)?

**Comment by travisghansen** [comment]
I got this working, seems the migration for the feeds.active column left all my feeds inactive.  Had to manually run the update SQL.
